graylog2-web-interface/src/components/common/ReactGridContainer.jsx

```diff
@@ -107,7 +115,39 @@ const ReactGridContainer = React.createClass({
     };
   },
 
+  getInitialState() {
+    return {
+      layout: this.computeLayout(this.props.positions),
+    };
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (!lodash.isEqual(nextProps.positions, this.props.positions)) {
+      this.setState({ layout: this.computeLayout(nextProps.positions) });
+    }
+  },
+
+  computeLayout(positions) {
+    return Object.keys(positions).map((id) => {
+      const { col, row, height, width } = positions[id];
+      return {
+        i: id,
+        x: col ? Math.max(col - 1, 0) : 0,
+        y: (row === undefined || row <= 0 ? Infinity : row - 1),
+        h: height || 1,
+        w: width || 1,
+      };
+    });
+  },
+
   _onLayoutChange(newLayout) {
+    // `onLayoutChange` may be triggered when clicking somewhere in a widget, check before propagating the change.
+    // Filter out additional Object properties in nextLayout, as it comes directly from react-grid-layout
+    const filteredNewLayout = newLayout.map(item => ({ i: item.i, x: item.x, y: item.y, h: item.h, w: item.w }));
+    if (lodash.isEqual(this.state.layout, filteredNewLayout)) {
```

Comment:
Would it make sense to move this kind of check into `shouldComponentUpdate` and keep the component stateless?

Reply:
I don't think so, as this callback is used by `react-grid-layout` to notify when a widget is dragged or resized and it's not modifying the state or deciding if the component should re-render. The check it's avoiding to call a prop when the layout didn't change.